### PR TITLE
Build manylinux wheels

### PR DESCRIPTION
**Pull request by EpicWink**
Build many-linux (2010) wheels in CI

PyPI requires uploaded wheels to be many-linux for distribution.

This PR incorporates #2 (so you might want to merge that in first).

PS: Travis CI is a hot piece of garbage. GitHub actions is so much easier to use.
You might want to squash commits during the merge.

### CODE REVIEW

**Comment by chaitan94** [comment]
You have beat me to it :sweat_smile:. I was working on the same, although I took a Github Actions and [cibuildwheel](https://github.com/joerick/cibuildwheel) based approach. I've recently set up such a workflow on one of my project https://github.com/adonmo/daylight/commit/ec413a9d8f10a024cbef8659362bd01b016d6654. Additionally, I refactored CMake build workflow to use a more modern pattern, avoiding the git submodules. That way I am able to build for macos (and possibly windows too - haven't tested this though). If you guys are interested I can raise a PR with my approach as well.

File: .travis.yml [hunk 1]
```diff
@@ -1,35 +1,52 @@
...
-  - export CXX=g++-8
-  - mkdir -p build
-  - cd build
-  - cmake ..
-  - make -j4
-  - make test
-  - cd ..
+jobs:
+  include:
+    - python: 2.7
```
Comment (bobfang1992) [inline .travis.yml:18]: I had a look at the most recent CI build on this branch and it seems the job is failing for Python 2.7 as I used some of the Python 3 features in my cpp code. (https://github.com/bobfang1992/pytomlpp/blob/master/src/pytomlpp.hpp#L39, timezone class in datetime module)

My suggestion is that we remove Python 2.7 support altogether. As it is not officially supported anyway and I would imagine people will migrate to Python 3. 

What's your thought?

**Comment by bobfang1992** [comment]
@EpicWink this is great. I was researching how to do this before but did not have any success. Let's drop Python 2.7 support and I will merge this :)

**Comment by bobfang1992** [comment]
> You have beat me to it 😅. I was working on the same, although I took a Github Actions and [cibuildwheel](https://github.com/joerick/cibuildwheel) based approach. I've recently set up such a workflow on one of my project [adonmo/daylight@ec413a9](https://github.com/adonmo/daylight/commit/ec413a9d8f10a024cbef8659362bd01b016d6654). Additionally, I refactored CMake build workflow to use a more modern pattern, avoiding the git submodules. That way I am able to build for macos (and possibly windows too - haven't tested this though). If you guys are interested I can raise a PR with my approach as well.

Hi @chaitan94, can you wait until we got this merged? and then you can submit your change in a separate PR on top of this one? Does this make sense? Thanks.

**Comment by bobfang1992** [comment]
@EpicWink @chaitan94 thanks for the contribution and advice. I will look into replacing Travis CI with Github Actions once we got both of your changes merged.

**Comment by chaitan94** [comment]
> Hi @chaitan94, can you wait until we got this merged? and then you can submit your change in a separate PR on top of this one? Does this make sense? Thanks.

Sure. I will push my changes to a forked repo and test there meanwhile. That way you guys can also have a look and compare. I can rebase it later after this gets merged.

File: .travis.yml (hunk 1 above)
Comment (EpicWink) [reply .travis.yml:18]: If this is to be used in packaging, you'll likely need to support Python 2.7 until January 2021. If you don't have Python 2, some packaging libraries can't use your module. This decision is up to you.

**Comment by EpicWink** [comment]
@chaitan94 looking forward to it. I'd like to see how cibuildwheel works

**Comment by chaitan94** [comment]
Added a PR: https://github.com/bobfang1992/pytomlpp/pull/4 can rebase later if required

File: .travis.yml (hunk 1 above)
Comment (bobfang1992) [reply .travis.yml:18]: Hi, what do you mean by packaging projects? I am working on this branch now to add py2 support.

**Comment by sinoroc** [comment]
If I am not mistaken for a dependency to be taken over in _packaging_ projects (_pip_, etc.) it should be pure Python, because otherwise it can't be vendored in, is that right? Making the argument for Python 2 compatibility irrelevant.

**Comment by EpicWink** [comment]
@bobfang1992 the force-push you just made removed all of my signed commits (and replaced them with your own commits). Did you want me to force-push all of my original commits (which are all identical) to keep the signature (the verified icon next to the commits in GitHub)? This won't matter if you intend to squash the commits

**Comment by EpicWink** [comment]
> If I am not mistaken for a dependency to be taken over in _packaging_ projects (_pip_, etc.) it should be pure Python, because otherwise it can't be vendored in, is that right? Making the argument for Python 2 compatibility irrelevant.

For pip's specific design goals, yes. Other projects may distribute themselves differently, for example usually I install poetry via pip, and conda is a whole thing.

**Comment by bobfang1992** [comment]
@EpicWink sorry I was just rebasing on #2, yeah please do, didn't expect to erase your commits.

**Comment by bobfang1992** [comment]
@EpicWink seems adding py2 support is not trivial and I think instead of me working on this branch, we can do this:

- let's remove py2.7 for now
- I will merge this
- and then I will create a new branch and work on that to add py2.7 support and create new PR 

Does this make sense? If you could remove py2.7 for now I will be merging it.

Thanks a lot!

**Comment by EpicWink** [comment]
Turns out I really want to squash those commits anyway. Should be good once the build finishes (however Python 2.7 is still broken, but that's out-of-scope for this PR. Another PR would either remove Python 2.7 support or fix the package for Python 2.7)

**Comment by bobfang1992** [comment]
@EpicWink  okay. I will merge this and remove Python2.7 on master and will work on a separate branch to add it back. Thanks for all the effort.

**Pull request merged**